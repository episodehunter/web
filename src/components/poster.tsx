import * as React from 'react'
import styled from 'styled-components'
import { images } from '../images.config'

type Props = {
  tvdbId: number
  zoom?: boolean
}
export const Poster = ({ tvdbId, zoom }: Props) => (
  <Wrapper zoom={zoom}>
    <Image src={images.poster.small(tvdbId)} />
  </Wrapper>
)

const Image = styled.img.attrs({
  src: (props: { src?: string }) => props.src
})`
  width: 100%;
`
const Wrapper = styled.div`
  position: relative;
  cursor: pointer;
  ${(props: { zoom?: boolean }) =>
    props.zoom
      ? `transition: 0.3s ease-in-out;
  &:hover {
    -webkit-transform: scale(1.05);
    -ms-transform: scale(1.05);
    transform: scale(1.05);
  }`
      : ''};
`
