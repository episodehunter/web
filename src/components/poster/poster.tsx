import React from 'react'
import { LazyLoadImage } from '@tjoskar/react-lazyload-img'

type Props = {
  width: number
  height: number
  imagePath: string
  style?: React.CSSProperties
}

export const Poster = ({ height, width, imagePath, style }: Props) => (
  <LazyLoadImage
    style={style}
    width={width}
    height={height}
    defaultImage="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw=="
    image={imagePath}
  />
)
