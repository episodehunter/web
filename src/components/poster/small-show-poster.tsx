import * as React from 'react'
import { images } from '../../images.config'
import { Poster } from './poster'

type Props = {
  tvdbId: number
  scale?: number
}

export const SmallShowPoster = ({ tvdbId, scale }: Props) => {
  const width = 185 * (scale || 1)
  const height = 273 * (scale || 1)
  return Boolean(tvdbId) ? (
    <Poster
      style={{ flexShrink: 0 }}
      width={width}
      height={height}
      imagePath={images.poster.small(tvdbId)}
    />
  ) : null
}
