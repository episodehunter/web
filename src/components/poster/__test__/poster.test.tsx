import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Poster } from '../poster'

it('Poster renders correctly', () => {
  const tree = renderer
    .create(<Poster width={100} height={200} imagePath={'some-path'} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
