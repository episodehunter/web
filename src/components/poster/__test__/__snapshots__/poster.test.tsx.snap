// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster renders correctly 1`] = `
<img
  defaultImage="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw=="
  height={200}
  image="some-path"
  width={100}
/>
`;
